pkg/eval/builtin_special.go

```diff
@@ -359,22 +366,43 @@ func use(fm *Frame, spec string, r diag.Ranger) (*Ns, error) {
 	if libDir == "" {
 		return nil, noSuchModule{spec}
 	}
-	return useFromFile(fm, spec, libDir+"/"+spec+".elv", r)
+	var path string
+	_, plugin := os.Stat(libDir + "/" + spec + ".so")
+	if os.IsNotExist(plugin) {
+		path = libDir + "/" + spec + ".elv"
+	} else {
+		path = libDir + "/" + spec + ".so"
+	}
+	return useFromFile(fm, spec, path, r)
 }
 
 // TODO: Make access to fm.Evaler.modules concurrency-safe.
 func useFromFile(fm *Frame, spec, path string, r diag.Ranger) (*Ns, error) {
-	if ns, ok := fm.Evaler.modules[path]; ok {
-		return ns, nil
-	}
-	code, err := readFileUTF8(path)
-	if err != nil {
-		if os.IsNotExist(err) {
+	if strings.HasSuffix(path, ".so") {
+		plug, err := plugin.Open(path)
+		if err != nil {
 			return nil, noSuchModule{spec}
 		}
-		return nil, err
+		sym, err := plug.Lookup("Ns")
+		if err != nil {
+			return nil, err
+		}
+		var ns **Ns
+		ns = sym.(**Ns)
+		return *ns, nil
+	} else {
+		if ns, ok := fm.Evaler.modules[path]; ok {
+			return ns, nil
+		}
```

Comment:
These two lines should remain at the beginning of the function - `.go` modules should be only loaded once too, like `.elv` modules.